presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/operator/aggregation/GeometryState.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.geospatial.operator.aggregation;
+
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.facebook.presto.spi.function.AccumulatorState;
+import com.facebook.presto.spi.function.AccumulatorStateMetadata;
+
+/**
+ * A state class for geometries.  Use this to save intermediate operations on geometry aggregations.
+ */
+@AccumulatorStateMetadata(stateSerializerClass = GeometryStateSerializer.class, stateFactoryClass = GeometryStateFactory.class)
+public interface GeometryState
+        extends AccumulatorState
+{
+    /**
+     * Returns the aggregate geometry
+     * @return the geometry
+     */
+    OGCGeometry getGeometry();
+
+    /**
+     * Sets the state's geometry, replacing the existing built state and recalculating its memory estimate
+     * @param geometry
+     */
+    void setGeometry(OGCGeometry geometry);
+
+    /**
+     * Takes the internal geometry and union's it with a separate geometry
+     * @param geometry the geometry to union this state's geometry with
+     */
+    void union(OGCGeometry geometry);
```

Comment:
Let's keep the state object simple, e.g. have it just provide set/getGeometry methods. The union operator will go into `UnionAgg`. This way this state object can be reused for other aggregation functions, e.g. `convex_full_agg`, `concave_hull_agg`, `centroid_agg` etc.